### PR TITLE
wally-cli: bump to 2.0.0

### PR DESCRIPTION
**Pull request by steveej**
#### Motivation for this change
New version supports new hardware.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/tools/wally-cli/default.nix [hunk 1]
```diff
@@ -1,28 +1,28 @@
-{ lib, buildGoPackage, fetchFromGitHub, pkg-config, libusb1 }:
+{ lib, buildGoModule, fetchFromGitHub, pkg-config, libusb1 }:
 
-buildGoPackage rec {
+buildGoModule rec {
   pname = "wally-cli";
-  version = "1.1.1";
+  version = "2.0.0-linux";
```
Comment (etu) [inline pkgs/development/tools/wally-cli/default.nix:5]: I find it a bit weird to have the platform in the version, but I see that they do that to their tags.
```suggestion
  version = "2.0.0";
```

File: pkgs/development/tools/wally-cli/default.nix [hunk 2]
```diff
@@ -1,28 +1,28 @@
...
 
   buildInputs = [ libusb1 ];
 
   src = fetchFromGitHub {
     owner = "zsa";
-    repo = "wally";
-    rev = "68960e452ee0f6c7142f5008d4b1cdc6284d3de7";
-    sha256 = "122m5v7s5wqlshyk2salmd848lqs4rrz54d2ap11ay61kijm0bs2";
+    repo = "wally-cli";
+    rev = version;
```
Comment (etu) [inline pkgs/development/tools/wally-cli/default.nix:17]: So to me it makes sense to set the tag properly without having the platform in the version.

```suggestion
    rev = "${version}-linux";
```

**Review by etu — APPROVED** [review]
Overall I approve these changes. From looking at hydra it didn't seem to build on darwin before either. So it wouldn't regress by starting to not build on darwin.

One might want to add `platforms = platforms.linux;` to the meta to make darwin an unsupported platform instead of just having failures.

**Comment by steveeJ** [comment]
Thanks @etu 

I incorporated your suggestions. For some reasons squashing your commits into my first one swallowed your co-authorship though :thinking:

**Comment by etu** [comment]
@steveeJ That's fine, thanks :slightly_smiling_face:

**Pull request merged**